include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
@gdr-at-ms I'm not sure what option I have left. `_HAS_STD_BYTE` is the symbol that indicates to us whether a user has opted into `std::byte` with MSVC STL or not. It is possible for people to opt-out to avoid conflicts with existing code (regardless of the version of their compiler). If we don't check that symbol in the MSVC case, then we would fail in those environments. At least this way people would get a fallback `gsl::byte` in those cases.